crates/bridge-exec/src/handler.rs

```diff
@@ -19,22 +20,22 @@ use strata_rpc_api::StrataApiClient;
 use strata_rpc_types::HexBytes;
 use tracing::{debug, info, warn};
 
-use crate::errors::{ExecError, ExecResult};
+use crate::{
+    errors::{ExecError, ExecResult},
+    ws_client::WsClientManager,
+};
 
 /// The execution context for handling bridge-related signing activities.
 #[derive(Clone)]
-pub struct ExecHandler<
-    L2Client: StrataApiClient + Sync + Send,
-    TxBuildContext: BuildContext + Sync + Send,
-> {
+pub struct ExecHandler<TxBuildContext: BuildContext + Sync + Send> {
     /// The build context required to create transaction data needed for signing.
     pub tx_build_ctx: TxBuildContext,
 
     /// The signature manager that handles nonce and signature aggregation.
     pub sig_manager: SignatureManager,
 
     /// The RPC client to connect to the RPC full node.
-    pub l2_rpc_client: L2Client,
+    pub l2_rpc_client_pool: Pool<WsClientManager>,
```

Comment:
Perhaps make a type alias like `type WsClientPool = Pool<WsClientManager>;` that can be re-exported from where `WsClientManager` is defined instead of having to make downstream consumers import the `Pool` type.